File: Oreons.Crud/Domain/Entities/Product.cs
using Shared.Entities;
using System;

namespace Domain.Entities
{
    public class Product : Entity
    {
        protected Product() { }
        public Product(string name, Guid subCategoryId)
        {
            Name = name;
            SubCategoryId = subCategoryId;
        }

        public string Name { get; set; }

        public virtual SubCategory SubCategory { get; set; }
        public Guid SubCategoryId { get; set; }

        public void Update(string name, Guid subCategoryId)
        {
            Name = name;
            SubCategoryId = subCategoryId;
        }
    }
}

File: front/src/app/models/product.ts
import { Subcategory } from "./subcategory";

interface Product {
  id: string | null,
  name: string,
  subCategoryId: string,
  subCategory: Subcategory | undefined | null
}

interface ProductListResponse {
  data: Array<Product>,
  success: boolean
}

interface ProductResponse {
  data: Product,
  success: boolean
}

export {
  Product,
  ProductListResponse,
  ProductResponse
}
